src/test/java/org/sagebionetworks/bridge/spring/controllers/EnrollmentControllerTest.java

```diff
@@ -0,0 +1,126 @@
+import org.mockito.Spy;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import org.sagebionetworks.bridge.TestUtils;
+import org.sagebionetworks.bridge.hibernate.HibernateEnrollment;
+import org.sagebionetworks.bridge.models.PagedResourceList;
+import org.sagebionetworks.bridge.models.accounts.UserSession;
+import org.sagebionetworks.bridge.models.studies.Enrollment;
+import org.sagebionetworks.bridge.services.EnrollmentService;
+
+public class EnrollmentControllerTest extends Mockito {
+    
+    @Mock
+    EnrollmentService mockService;
+    
+    @Mock
+    HttpServletRequest mockRequest;
+    
+    @Mock
+    HttpServletResponse mockResponse;
+    
+    @InjectMocks
+    @Spy
+    EnrollmentController controller;
+    
+    @Captor
+    ArgumentCaptor<Enrollment> enrollmentCaptor;
+    
+    @BeforeMethod
+    public void beforeMethod() {
+        MockitoAnnotations.initMocks(this);
+        doReturn(mockRequest).when(controller).request();
+        doReturn(mockResponse).when(controller).response();
+        
+        UserSession session = new UserSession();
+        session.setAppId(TEST_APP_ID);
+        doReturn(session).when(controller).getAuthenticatedSession(RESEARCHER, ADMIN);
+    }
+    
+    @Test
+    public void getEnrollmentsForStudy() {
+        Enrollment en1 = Enrollment.create(TEST_APP_ID, TEST_STUDY_ID, "user1");
+        Enrollment en2 = Enrollment.create(TEST_APP_ID, TEST_STUDY_ID, "user2");
+        PagedResourceList<Enrollment> page = new PagedResourceList<>(ImmutableList.of(en1, en2), 10);
+        when(mockService.getEnrollmentsForStudy(TEST_APP_ID, TEST_STUDY_ID, ENROLLED, 5, 40)).thenReturn(page);
+        
+        PagedResourceList<Enrollment> retValue = controller.getEnrollmentsForStudy(
+                TEST_STUDY_ID, "5", "40", "enrolled");
+        assertSame(retValue, page);
+        
+        verify(mockService).getEnrollmentsForStudy(TEST_APP_ID, TEST_STUDY_ID, ENROLLED, 5, 40);
+    }
+    
+    @Test
+    public void enroll() throws Exception {
+        Enrollment enrollment = new HibernateEnrollment();
+        enrollment.setEnrolledOn(CREATED_ON);
+        enrollment.setExternalId("anExternalId");
+        enrollment.setAccountId(USER_ID);
+        enrollment.setConsentRequired(true);
+        TestUtils.mockRequestBody(mockRequest, enrollment);
+        
+        Enrollment completed = new HibernateEnrollment();
+        when(mockService.enroll(any())).thenReturn(completed);
+        
+        Enrollment retValue = controller.enroll(TEST_STUDY_ID);
+        assertSame(retValue, completed);
+        
+        verify(mockService).enroll(enrollmentCaptor.capture());
+        
+        Enrollment value = enrollmentCaptor.getValue();
+        assertEquals(value.getEnrolledOn(), CREATED_ON);
+        assertEquals(value.getExternalId(), "anExternalId");
+        assertEquals(value.getAccountId(), USER_ID);
+        assertTrue(value.isConsentRequired());
```

Comment:
Should we assert that we added App and Study?

Reply:
Added